### PR TITLE
[MIG][12.0] backport base_changeset

### PR DESCRIPTION
**Pull request by hbrunn**

### CODE REVIEW

**Comment by hbrunn** [comment]
he @OCA-git-bot, this addon [has a maintainer](https://github.com/OCA/server-tools/pull/2167/commits/fb135a3236bf0a442762e88678fa3048df204415#diff-d342be9a2d196bbde132681a41efce8147ffdd1c9d4894b08b26f4a5186e7331R10) - are you reading the files wrongly?

**Comment by hbrunn** [comment]
@astirpe @dsolanki-initos I think commits b6998d0 and ea1c0ce are relevant for later versions too, you should be able to just cherry-pick them

**Comment by astirpe** [comment]
Thank you @hbrunn !

**Review by astirpe — APPROVED** [review]

**Comment by StefanRijnhart** [comment]
Looks like this module's tests fail
```
2021-10-28 14:27:54,492 7531 ERROR openerp_test odoo.sql_db: bad query: INSERT INTO "mail_followers" ("id", "partner_id", "res_id", "res_model") VALUES (nextval('mail_followers_id_seq'), 2, 82, 'res.partner') RETURNING id

ERROR: duplicate key value violates unique constraint "mail_followers_mail_followers_res_partner_res_model_id_uniq"

DETAIL:  Key (res_model, res_id, partner_id)=(res.partner, 82, 2) already exists.

2021-10-28 14:27:54,504 7531 ERROR openerp_test odoo.addons.base_changeset.tests.test_changeset_flow: ERROR
```

**Comment by hbrunn** [comment]
now without breaking when mail is installed

File: base_changeset/models/record_changeset.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        readonly=True,
+    )
+    change_ids = fields.One2many(
+        comodel_name="record.changeset.change",
+        inverse_name="changeset_id",
+        string="Changes",
+        readonly=True,
+    )
+    date = fields.Datetime(
+        string="Modified on", default=fields.Datetime.now(), index=True, readonly=True
```
Comment (StefanRijnhart) [inline base_changeset/models/record_changeset.py]: Could it be that the callable was intended?
```suggestion
        string="Modified on", default=fields.Datetime.now, index=True, readonly=True
```

**Comment by astirpe** [comment]
@hbrunn maybe it is worth to backport https://github.com/OCA/server-tools/pull/2184 within this PR as well, what do you think?

**Comment by hbrunn** [comment]
I totally think so, but I'm not sure if this should be in this PR or in a subsequent one

File: base_changeset/models/record_changeset.py (hunk 1 above)
Comment (hbrunn) [reply base_changeset/models/record_changeset.py]: oh yes, this seems wrong, thanks! @astirpe are we missing something fancy?

**Comment by hbrunn** [comment]
@StefanRijnhart fixed your last point + squashed commits

**Review by StefanRijnhart — APPROVED** [review]
Thanks!

**Pull request merged**